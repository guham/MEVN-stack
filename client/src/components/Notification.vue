<!-- eslint-disable max-len -->
<template>
  <div class="fixed pin-b pin-x bg-white p-4 text-grey-darkest border-t border-grey-light">
    <div class="flex w-full justify-between items-center">
      <div class="hidden lg:flex" />
      <div class="flex mr-6">
        <span
          :class="{ 'text-red': notification.type === 'error' }"
          class="mr-2"
        >
          {{ $t(notification.type) }}
        </span>{{ $t(notification.text) }}
      </div>
      <div class="flex self-start">
        <button
          class="flex items-center px-3 py-2 text-grey-dark"
          aria-label="Close notification"
          @click="removeNotification()"
        >
          <svg
            class="fill-current h-4 w-4"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 8.586L2.929 1.515 1.515 2.929 8.586 10l-7.071 7.071 1.414 1.414L10 11.414l7.071 7.071 1.414-1.414L11.414 10l7.071-7.071-1.414-1.414L10 8.586z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</template>

<i18n>
{
  "en": {
    "error": "Error",
    "%SOMETHING_WENT_WRONG%": "Something went wrong. Please try again."
  },
  "fr": {
    "error": "Erreur",
    "%SOMETHING_WENT_WRONG%": "Une erreur est survenue. Merci de réessayer."
  },
  "es": {
    "error": "Error",
    "%SOMETHING_WENT_WRONG%": "Ha ocurrido un error. Por favor, inténtalo de nuevo."
  }
}
</i18n>

<script>
export default {
  name: 'Notification',
  props: {
    notification: {
      type: Object,
      required: true,
    },
  },

  methods: {
    removeNotification() {
      this.$emit('removeNotification');
    },
  },
};
</script>
