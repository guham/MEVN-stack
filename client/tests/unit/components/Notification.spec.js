import NotificationComponent from '@/components/Notification.vue';
import Notification from '@/models/notification';
import factory from '../factory';

beforeEach(() => {
  jest.resetModules();
  jest.clearAllMocks();
});

describe('Notification.vue', () => {
  test('has the expected html structure', () => {
    const notification = new Notification();
    const wrapper = factory(NotificationComponent, {}, { notification });
    const template = wrapper.html();
    expect(template).toMatchSnapshot();
  });

  test('emit `removeNotification` event when Close button is clicked', (done) => {
    const notification = new Notification();
    const wrapper = factory(NotificationComponent, {}, { notification });
    wrapper.find('button').trigger('click');
    wrapper.vm.$nextTick(() => {
      expect(wrapper.emitted('removeNotification')).toBeTruthy();
      expect(wrapper.emitted('removeNotification').length).toBe(1);
      expect(wrapper.emitted('removeNotification')[0]).toEqual([]);
      done();
    });
  });
});
