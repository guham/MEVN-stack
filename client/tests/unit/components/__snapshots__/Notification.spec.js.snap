// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification.vue has the expected html structure 1`] = `
<div class="fixed pin-b pin-x bg-white p-4 text-grey-darkest border-t border-grey-light">
  <div class="flex w-full justify-between items-center">
    <div class="hidden lg:flex"></div>
    <div class="flex mr-6"><span class="mr-2">
        info
      </span>
    </div>
    <div class="flex self-start"><button aria-label="Close notification" class="flex items-center px-3 py-2 text-grey-dark"><svg viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg" class="fill-current h-4 w-4">
          <path d="M10 8.586L2.929 1.515 1.515 2.929 8.586 10l-7.071 7.071 1.414 1.414L10 11.414l7.071 7.071 1.414-1.414L11.414 10l7.071-7.071-1.414-1.414L10 8.586z"></path>
        </svg></button></div>
  </div>
</div>
`;
